modules/ephemeral-storage-additional-details-about-volumeattributes-on-shared-resource-pod-volumes.adoc

```diff
@@ -0,0 +1,38 @@
+:_mod-docs-content-type: REFERENCE
+
+[id="ephemeral-storage-additional-details-about-volumeattributes-on-shared-resource-pod-volumes_{context}"]
+= Additional details about VolumeAttributes on shared resource pod volumes
+
+
+[role="_abstract"]
+The following attributes affect shared resource pod volumes in various ways:
+
+* The `refreshResource` attribute in the `volumeAttributes` properties.
+* The `refreshResources` attribute in the Shared Resource CSI Driver configuration.
+* The `sharedSecret` and `sharedConfigMap` attributes in the `volumeAttributes` properties.
+
+== The `refreshResource` attribute
+
+The Shared Resource CSI Driver honors the `refreshResource` attribute in `volumeAttributes` properties of the volume. This attribute controls whether updates to the contents of the underlying `Secret` or `ConfigMap` object are copied to the volume after the volume is initially provisioned as part of pod startup. The default value of `refreshResource` is `true`, which means that the contents are updated.
+
+[IMPORTANT]
+====
+If the Shared Resource CSI Driver configuration has disabled the refreshing of both the shared `SharedSecret` and `SharedConfigMap` custom resource (CR) instances, then the `refreshResource` attribute in the `volumeAttribute` properties has no effect. The intent of this attribute is to disable refresh for specific volume mounts when refresh is generally allowed.
+====
+
+== The `refreshResources` attribute
+
+You can use a global switch to enable or disable refreshing of shared resources. This switch is the `refreshResources` attribute in the `csi-driver-shared-resource-config` config map for the Shared Resource CSI Driver, which you can find in the `openshift-cluster-csi-drivers` namespace. If you set this `refreshResources` attribute to `false`, none of the `Secret` or `ConfigMap` object-related content stored in the volume is updated after the initial provisioning of the volume.
+
+[IMPORTANT]
+====
+Using this Shared Resource CSI Driver configuration to disable refreshing affects all the cluster's volume mounts that use the Shared Resource CSI Driver, regardless of the `refreshResource` attribute in the `volumeAttributes` properties of any of those volumes.
+====
+
+== Validation of volumeAttributes before provisioning a shared resource volume for a pod
+
+In the `volumeAttributes` of a single volume, you must set either a `sharedSecret` or a `sharedConfigMap` attribute to the value of a `SharedSecret` or a `SharedConfigMap` CR instance. Otherwise, when the volume is provisioned during pod startup, a validation checks the `volumeAttributes` of that volume and returns an error to the kubelet under the following conditions:
```

Comment:
```suggestion
In the `volumeAttributes` of a single volume, you must set either a `sharedSecret` or a `sharedConfigMap` attribute to the value of a `SharedSecret` or a `SharedConfigMap` instance. Otherwise, when the volume is provisioned during pod startup, a validation checks the `volumeAttributes` of that volume and returns an error to the kubelet under the following conditions:
```